lib/validation.js

```diff
@@ -0,0 +1,64 @@
+const { INTERVAL_SHORTCUTS } = require('./utils');
+
+class LimitdRedisValidationError extends Error {
+  constructor(msg, extra) {
+    super();
+    this.name = this.constructor.name;
+    this.message = msg;
+    Error.captureStackTrace(this, this.constructor);
+    if (extra) {
+      this.extra = extra;
+    }
+  }
+}
+
+function validateParams(params, buckets) {
+  if (typeof params !== 'object') {
+    return new LimitdRedisValidationError('params are required', { code: 101 });
+  }
+
+  if (typeof params.type !== 'string') {
+    return new LimitdRedisValidationError('type is required', { code: 102 });
+  }
+
+  if (typeof buckets[params.type] === 'undefined') {
+    return new LimitdRedisValidationError(`undefined bucket type ${params.type}`, { code: 103 });
+  }
+
+  if (typeof params.key !== 'string') {
+    return new LimitdRedisValidationError('key is required', { code: 104 });
+  }
+
+  if (typeof params.configOverride !== 'undefined') {
+    try {
+      validateOverride(params.configOverride);
+    } catch (error) {
+      return error;
+    }
+  }
+}
+
+function validateOverride(configOverride) {
+  if (typeof configOverride !== 'object') {
+    throw new LimitdRedisValidationError('configuration overrides must be an object', { code: 105 });
+  }
+
+  // If size is provided, nothing more is strictly required
+  // (as in the case of static bucket configurations)
+  if (typeof configOverride.size === 'number') {
+    return;
+  }
+
+  const interval = Object.keys(configOverride)
```

Comment:
We would still need to validate is a valid interval if `size` is provided.

Reply:
Would we?  I thought this worked similar to the way LimitD worked: a `size` param with no intervals simply meant it was a bucket that did not get any "drips", such as was the case with single-user brute force (when it used LimitD).

For instance, this is a valid bucket configuration (see [db.tests.js](https://github.com/auth0/limitd-redis/blob/2b44821a2f6bc0ad80a778310367483a0d2f7d9f/test/db.tests.js#L36-L38)):
```
      '8.8.8.8': {
        size: 10
      }
```

I was working off of the understanding that `configOverride` parameters would have the same contract as the bucket configuration overrides.  Should we instead validate an interval is present only on overrides?

Reply:
Ok, right, since your goal is only checking for presence, then I guess it's fine. On `size` we also validated type, hence the confusion.

But let's not get too deep into these validations; they are lacking in general.